lib.py

```diff
@@ -47,21 +44,19 @@ async def benchmark(
                 return
 
             with Database() as db:
-                # TODO dont hardcode 2023
-                answers_map = db.load_answers(2023, day, part)
+                answers_map = db.load_answers(year, day, part)
 
-            for in_file in get_input_files(day):
-                logger.info("Processing file: %s", in_file)
-                load_input(tmpdir, day, in_file)
-                result_lst = await run_code(op_name, op_id, tmpdir, in_file)
-                result = process_run_result(in_file, answers_map, result_lst)
-                if result is not None:
-                    results.append(result)
+                for in_file, contents in db.get_inputs(year, day).items():
+                    logger.info("Processing file: %s", in_file)
+                    load_input(tmpdir, contents)
+                    result_lst = await run_code(op_name, op_id, tmpdir, in_file)
+                    result = process_run_result(in_file, answers_map, result_lst)
+                    if result is not None:
+                        results.append(result)
 
             if results:
                 with Database() as db:
-                    # TODO dont hardcode 2023
-                    db.save_results(op_id, 2023, day, part, code, results)
+                    db.save_results(op_id, year, day, part, code, results)
```

Comment:
This probably should be in the same `with` block as the above code going forward, unless we're going to work harder to separate DB code from stuff that's just waiting for Docker to run.

Reply:
yeah, the `if results:` seems to not do anything meaningful either